src/pages/index.js

```diff
@@ -31,56 +63,114 @@ const popupEditProfileForm = new PopupWithForm(
   ".modal_profile-edit",
   handleProfileEditSubmit
 );
-const gallerySection = new Section(
-  { items: initialCards, renderer: cardRenderer },
-  cardGalleryEL
+const popupAvatarForm = new PopupWithForm(
+  ".modal_change-avatar",
+  handleAvatarFormSubmit
 );
-
-//////////////////////////  functions  //////////////////////////
+////////////////////////////////////////////////////////////
+///////////////////////   API    ///////////////////////////
+////////////////////////////////////////////////////////////
+const apiCall = new Api(
+  "https://around-api.en.tripleten-services.com/v1",
+  serverToken
+);
+////////////////////////////////////////////////////////////
+///////////////////////   FUNCTIONS    /////////////////////
+////////////////////////////////////////////////////////////
+function handleDeleteFormSubmit(data) {
+  apiCall
+    .deleteCard(data.data)
+    .then(data.cardElement.remove())
+    .catch((err) => {
+      console.error(err);
+    });
+}
+function handleAvatarFormSubmit(data) {
+  apiCall
+    .patchAvatar(data)
+    .then((data) => {
+      user.setUserAvatar(data);
+      popupAvatarForm.close();
+      avatarForm.reset();
+      popupAvatarForm.renderSaving(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.